From 85cdda24ebd68a2127e57b38faf790572b9e6bee Mon Sep 17 00:00:00 2001
From: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Date: Wed, 24 Apr 2024 16:09:56 +0200
Subject: [PATCH] dt-bindings: remoteproc: add Cortex-M0 support

Add new description for the support of the Cortex-M0.

Change-Id: I178c7eafc77e382db5846ed4b6bdbd56b276bd72
Signed-off-by: Gwenael Treuveur <gwenael.treuveur@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/376630
Tested-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
Domain-Review: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
Reviewed-by: Arnaud POULIQUEN <arnaud.pouliquen@st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
---
 .../bindings/remoteproc/st,stm32-m0.yaml      | 106 ++++++++++++++++++
 1 file changed, 106 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/remoteproc/st,stm32-m0.yaml

diff --git a/Documentation/devicetree/bindings/remoteproc/st,stm32-m0.yaml b/Documentation/devicetree/bindings/remoteproc/st,stm32-m0.yaml
new file mode 100644
index 000000000000..36fd06e47b62
--- /dev/null
+++ b/Documentation/devicetree/bindings/remoteproc/st,stm32-m0.yaml
@@ -0,0 +1,106 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/remoteproc/st,stm32-m0.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: STMicroelectronics STM32 remote processor controller for Cortex-M0+
+
+description:
+  This document defines the binding for the remoteproc component that loads and
+  boots firmwares on the M0 of the smartrun domain on the ST32MP2 series.
+
+maintainers:
+  - Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
+
+properties:
+  compatible:
+    enum:
+      - st,stm32mp2-m0
+    description:
+      Use "st,stm32mp2-m0" for the stm32mp2 Cortex-M0 coprocessor management by Linux
+  reg:
+    maxItems: 1
+
+  clocks:
+    minItems: 1
+
+  clock-names:
+    items:
+      - const: c3_clk
+
+  resets:
+    minItems: 1
+
+  reset-names:
+    items:
+      - const: mcu_rst
+
+  interrupts:
+    description: Should contain the IWDG5 watchdog reset interrupt
+    maxItems: 1
+
+  wakeup-source: true
+
+  mboxes:
+    description:
+      This property is required only if the rpmsg functionality is used.
+    items:
+      - description: |
+          A channel (a) used to communicate with the remote proc.
+          Bi-directional channel:
+            - from local to remote = send message
+            - from remote to local = send message ack
+      - description: |
+          A channel (b) working the opposite direction of channel (a)
+      - description: |
+          A channel (c) used by the local proc to notify the remote proc that it
+          is about to be shut down.
+          Unidirectional channel:
+            - from local to remote, where ACK from the remote means that it is
+              ready for shutdown
+    minItems: 1
+
+  mbox-names:
+    items:
+      - const: tx
+      - const: rx
+      - const: shutdown
+    minItems: 1
+
+  memory-region:
+    description:
+      List of phandles to the reserved memory regions associated with the
+      remoteproc device. This is variable and describes the memories shared with
+      the remote processor (e.g. remoteproc firmware, shared memory for
+      interprocessor communication ...).
+      (see ../reserved-memory/reserved-memory.txt)
+
+  st,auto-boot:
+    $ref: /schemas/types.yaml#/definitions/flag
+    description:
+      If defined, when remoteproc is probed, it loads the default firmware and
+      starts the remote processor.
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - clock-names
+  - resets
+  - reset-names
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/reset/st,stm32mp25-rcc.h>
+    m0_rproc: m0@0 {
+      compatible = "st,stm32mp2-m0";
+      reg = <0 0>;
+      clocks = <&rcc CK_CPU3>;
+      clock-names = "c3_clk";
+      resets = <&rcc C3_R>;
+      reset-names = "mcu_rst";
+    };
+...
-- 
2.39.5

